src/hw-monitor.cpp

```diff
@@ -223,4 +225,115 @@ namespace librealsense
         librealsense::copy(&value, gvd.data() + offset, 1);
         return value;
     }
+
+    int hw_monitor_extended_buffers::get_msg_length(command cmd) const
+    {
+        int msg_length = HW_MONITOR_BUFFER_SIZE;
+        if (cmd.cmd == ds::fw_cmd::GET_HKR_CONFIG_TABLE || cmd.cmd == ds::fw_cmd::SET_HKR_CONFIG_TABLE)
+        {
+            switch (cmd.param2)
+            {
+            case static_cast<int>(ds::ds6_calibration_table_id::coefficients_table_id):
+            {
+                msg_length = sizeof(ds::ds6_coefficients_table);
+                break;
+            }
+            case static_cast<int>(ds::ds6_calibration_table_id::rgb_calibration_id):
+            {
+                msg_length = sizeof(ds::ds6_rgb_calibration_table);
+                break;
+            }
+            default:
+                throw std::runtime_error(rsutils::string::from() << "command GET_HKR_CONFIG_TABLE with wromg param2");
+            }
+        }
+        return msg_length;
+    }
+
+    int hw_monitor_extended_buffers::get_number_of_chunks(int msg_length) const
+    {
+        float ratio = static_cast<float>(msg_length) / 1024;
+        int ratio_int = static_cast<int>(ratio);
+        float res = ratio - ratio_int;
+        return ratio_int + ((res > 0.0f) ? 1 : 0);
+    }
+
+
+    std::vector<uint8_t> hw_monitor_extended_buffers::send(command cmd, hwmon_response* p_response, bool locked_transfer) const
+    {
+        //int send_msg_length = get_send_msg_length(cmd);
+        //int send_chunks = send_msg_length / 1024;
+
+        int recv_msg_length = get_msg_length(cmd);
+        uint16_t recv_chunks = get_number_of_chunks(recv_msg_length);
+
+        std::vector<byte> recv_msg;
+
+        for (int i = 0; i < recv_chunks; ++i)
+        {
+            // updating the chunk number in case several chunks are expected
+            if (recv_chunks > 1)
```

Comment:
If it's 1 meaning <= 1024 bytes should we place 0 in the MSB? not 1?
isn't it num_of_chunks | chunk_num?
here I see:
1000 bytes --> 0 | 0
1500 bytes --> 2 | 0
                         2 | 1

Reply:
This PR does not have the actual splitting to chunks, right? (both ways)
just making sure I understand

Reply:
> If it's 1 meaning <= 1024 bytes should we place 0 in the MSB? not 1? isn't it num_of_chunks | chunk_num? here I see: 1000 bytes --> 0 | 0 1500 bytes --> 2 | 0 2 | 1

Done - great you caught it :-)



> This PR does not have the actual splitting to chunks, right? (both ways) just making sure I understand

Right - only starting of implementation is here - it can be removed from this PR if you prefer. But it has no bad impact (tested only with less than 1KB files)